File: day-85/_298_Reverse_an_array.cpp
// _298_Reverse_an_array.cpp

void swap(int &a, int &b){
    a = a ^ b;
    b = a ^ b;
    a = a ^ b;
}

void reverseArr(vector<int> &arr)
{
    int start = 0;
    int end = arr.size() - 1;
    while (start < end)
    {
        swap(arr[start], arr[end]);
        start++;
        end--;
    }
}

// Original array: 1 2 3 4 5 
// Reversed array: 5 4 3 2 1

File: day-85/_298_Reverse_an_array.py
# _298_Reverse_an_array.py
    
def reverse(arr):
    start = 0
    end = (len(arr))-1
    while(start<end):
        arr[start],arr[end] = arr[end],arr[start]
        start += 1
        end -=1

arr = [10,12,13,14,5]
reverse(arr)
print(arr)

# Original array: 1 2 3 4 5 
# Reversed array: 5 4 3 2 1